pkg/sidecars/server/server.go

```diff
@@ -0,0 +1,115 @@
+package sidecarserv
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/inspr/inspr/pkg/sidecars/models"
+)
+
+// Server is a struct that contains the variables necessary
+// to handle the necessary routes of the rest API
+type Server struct {
+	broker       string
+	Reader       models.Reader
+	Writer       models.Writer
+	inAddr       string
+	outAddr      string
+	client       *http.Client
+	runningRead  bool
+	runningWrite bool
+}
+
+// NewServer returns a new sidecar server
+func NewServer() *Server {
+	return &Server{}
+}
+
+// Init - configures the server
+func (s *Server) Init(r models.Reader, w models.Writer, broker string) {
+
+	// server fetches addresses variable names from models.
+	envVars := models.GetSidecarConnectionVars(broker)
+	if envVars == nil {
+		panic(fmt.Sprintf("%s broker's enviroment variables not configured", broker))
+	}
+	s.broker = broker
+
+	// server fetches required addresses from deployment.
+	inAddr, ok := os.LookupEnv(envVars.WriteEnvVar)
+	if !ok {
+		panic(fmt.Sprintf("[ENV VAR] %s not found", envVars.WriteEnvVar))
+	}
+
+	outAddr, ok := os.LookupEnv(envVars.ReadEnvVar)
+	if !ok {
+		panic(fmt.Sprintf("[ENV VAR] %s not found", envVars.ReadEnvVar))
+	}
+
+	s.inAddr = fmt.Sprintf(":%s", inAddr)
+	s.outAddr = fmt.Sprintf("http://localhost:%v", outAddr)
+	s.client = &http.Client{}
+
+	// implementations of write and read for a specific sidecar
+	s.Reader = r
+	s.Writer = w
+} // looked
+
+// Run starts the server on the port given in addr
+func (s *Server) Run(ctx context.Context) {
```

Comment:
It's a good idea to make this function return an error. So instead of using `log.Fatal` throughout the function, you return an error after `gracefulShutdown`.  
Also, change the logs to use `zap.logger`.